### DISCUSSION TITLE
Ideas to define higher management view on OSS effort for OSPO strategy

### DISCUSSION THREAD

**Discussion by anajsana** [todogroup/ospology#106] [OSPO Discussions]
Question raised by Matthias via [slack](https://thetodogroup.slack.com/archives/C0GBYUWCB/p1652272291465699):

*Hi folks. I have a request for other OSPO members regarding their OSS strategy. Within our own OSPO, we've come out with few sentences that I'd like to share with you to see if they make sense... and maybe also discover what other OSPOs have come up with. The goal is to shortly and clearly express the higher management view on our OSS effort, and from that we could derive all our policies and processes. Here it is:*

> Open source software is an essential part of our technology landscape.
We use it to accelerate our pace of innovation, stay close to industry standards and rely on well-tested technologies.
While our goal is not to become a major open source contributor, we contribute back to open-source in cases where it allows us to advance our strategy.

*hint: we're a financial institution, not a software company*

**Comment by anajsana** [comment]
@byjrack replied:

*I might say remove the couching language as it sort of boxes you in a bit.  I tend to emphasize collaboration as the common theme and engaging in open source projects or foundation working groups as places where that happens.  And this is coming from a financial as well.*

**Reply by anajsana** [reply]
Matthias replied:

*Thanks for the suggestion, @byjrack how's yours worded, I'm curious 🙂*

**Reply by anajsana** [reply]
@byjrack replied:

*I am sure it changes every time since it is more a speaker note than some repeated phrase, but something like “Collaboration is the foundation of modern software engineering and IT practices.  Open Source, StackOverflow, Apache Foundation are all places we engage with others with the goal to do great things more effectively.  Our associates expect to be able to engage in all of these spaces and it is our responsibility as leaders to enable that through intelligent guardrails and responsible policies.”  That would sort of be my elevator speech for an OSPO or a focus on Developer Effectiveness.  That then can jump off into a number of threads tuned to the audience.*

**Reply by anajsana** [reply]
Matthias replied:

*Looks great indeed, thanks for the suggestion! (it somehow shows we're not native English speakers)*

**Reply by anajsana** [reply]
@byjrack replied:

*sometimes folks' brains race to the source code and get lost in copyrights and intellectual property aspects which while important is more of an outcome of the collaboration thing.  and often it's the collaboration and less the lines of code leaders may have a hard time getting their heads around.*

*💯 my pitch is tuned to a western mindset for a mature financial services company based on where I want them to be in the future.  looking at your audience’s local context and what motivates them is always a key part of any discussion.*

**Comment by anajsana** [comment]
@anajsana replied:

*Maybe you might find some inspiration from Goldman Sachs Open Source website. They recently made some edits to it 🙂  https://developer.gs.com/discover/open-source*

**Reply by anajsana** [reply]
Matthias replied:

*Thanks @anajsana I will try to find something there as well*

**Comment by anajsana** [comment]
Matt replied:

*This is such a fun and important topic. If there’s interest, I’m happy to facilitate a meeting if folks would like to weigh in on how to design ospo metrics, and then turn the discussion into a blog post*

**Comment by anajsana** [comment]
@DuaneOBrien replied:

*It's a somewhat stylistic preference, but I would recommend you aim to distill the strategy down to a single short sentence that doesn't use the word And, and separate it from a description of your context or any vision statements.*

**Reply by anajsana** [reply]
Matthias replied:

*I agree that should be super high level, but still giving a clear vision*

**Reply by anajsana** [reply]
@DuaneOBrien replied:

*To show you what I mean:*

*`Open source software is an essential part of our technology landscape.` - I see how this phrase helps us all get on the same page, and it helps describe where we are but doesn't articulate where we're going.*

*`We use it to accelerate our pace of innovation, stay close to industry standards and rely on well-tested technologies.` - This feels close to a Vision statement, where you're imagining the world as you want it to be. It helps set the context for the discussion, but it also doesn't articulate where we're going.*

*`While our goal is not to become a major open source contributor, we contribute back to open-source in cases where it allows us to advance our strategy.` I think your Strategy statement is in here. This tells me where you want to go. You want to focus on a few, well-placed contributions that align with your business.*

*I think of the framing a little more like "Where do we want to be?" (Mission), "What will it look like when we get there?" (Vision), and "How will we get there?" (Strategy). The individual sentences are not often used in isolation from each other, but focusing on a single sentence articulation of each makes it easier to keep them distinct from each other.*

*Again, this is all stylistic preference. Wordsmithing is hard.*

**Reply by anajsana** [reply]
Matthias replied:

*Thanks Duane, I'll submit this to the team*

**Comment by anajsana** [comment]
@justinabrahms replied:

*"stay close to industry standards" is a good phrase. Something that's a big focus for us with our OSPO*

**Reply by anajsana** [reply]
@DuaneOBrien replied:

*To be contrary to myself - I'm a big fan of "Good enough to get us there for now" - If your current strategy articulation gets you running toward the right work, then you can always sharpen the language over time.*

**Reply by anajsana** [reply]
Matthias replied:

*Thanks for your answers 😄*

**Comment by awright** [comment]
I find much value in this thread and would be interested in the moderated session matt offered to facilitate.

**Comment by hyandell** [comment]
> We use it to accelerate our pace of innovation, stay close to industry standards and rely on well-tested technologies.

Just a note on the missing oxford comma in case that's part of your editorial standards.

> While our goal is not to become a major open source contributor, we contribute back to open-source in cases where it allows us to advance our strategy.

For me, 'advance our strategy' is a little too bond villain there. Do you have one strategy? Or a strategy per project you're contributing to? I also suspect you don't have a goal that says "Whatever we do, avoid becoming a major open source contributor".

How about something along the lines of:

_While we don't have a goal to become a major open source contributor, the success of the projects we rely on is in our best interest and we seek to support their success._ 

I've found that "bugfixes to what we use" is an easy first hurdle for a business to accept, and my aim above is to encourage folk within your company to focus on 'success' to explain why a contribution will add value to the project, and for 'seek' to provide the fuzziness that this is an aspiration, not an every time guarantee.
